test/Serilog.Settings.Configuration.Tests/ConfigurationSettingsTests.cs

```diff
@@ -556,5 +559,43 @@ public void WriteToSubLoggerWithLevelSwitchIsSupported()
 
             Assert.Equal(1, DummyRollingFileSink.Emitted.Count);
         }
+
+        [Trait("Bugfix", "#103")]
+        [Fact]
+        public void MultipleArgumentValuesThrowsInvalidOperationException()
+        {
+            var jsonDiscreteValue = @"{
+                ""Serilog"": {            
+                    ""Using"": [""TestDummies""],
+                    ""WriteTo"": [{
+                        ""Name"": ""DummyRollingFile"",
+                        ""Args"": {""pathFormat"" : ""C:\\""}
+                    }]        
+                }
+            }";
+
+            var jsonComplexValue = @"{
+                ""Serilog"": {            
+                    ""Using"": [""TestDummies""],
+                    ""WriteTo"": [{
+                        ""Name"": ""DummyRollingFile"",
+                        ""Args"": {""pathFormat"" : { ""foo"" : ""bar"" } }
+                    }]        
+                }
+            }";
+
+            // These will combine into a ConfigurationSection object that has both
+            // Value == "C:\" and GetChildren() == List<string>. No configuration
+            // extension matching this exists (in theory an "object" argument could
+            // accept either value). ConfigurationReader should throw as soon as
+            // the multiple values are recognized; it will never attempt to locate
+            // a matching argument.
+
+            var ex = Assert.Throws<InvalidOperationException>(()
+                => ConfigFromJson(jsonDiscreteValue, jsonComplexValue));
```

Comment:
think `=>` shoudl go prev line ?

Reply:
I meant that the `=>` rarely starts a line in normal code and instead goes on prev line
i.e. the full `() =>` on prev line